src/macros.jl

```diff
@@ -595,69 +602,94 @@ end
+(which we simply call *terms* hereafter), as in:
 
-As an exception to the rule, if the right-hand side has the form `A*x + B*u` or
-`A*x + B*u + c` the equation is parsed as a controlled linear (affine) system
-even though the `input` variable does not correspond to the value of `u`.
+```julia
+@system(dynamic eq., set constr., input constr., input specif., noise spec., dimens.)
+```
 
-If the left-hand side corresponds to a multiplicative term in the form `E*x⁺`
-or `E*x'`, the equation is parsed as an algebraic system.
+The different terms that compose the system's definition do not have to appear in
+any particular order. Moreover, the only mandatory term is the dynamic equation;
+the other terms are optional and default values may apply depending on the system
+type; this is explained next.
+
+**Dynamic equation.** The time derivative in a continuous system is specified
+by using `'`, as in `x' = A*x`. A discrete system is specified using `⁺` (which
+can be written with the combination of keys `\\^+[TAB]`), as in `x⁺ = A*x`.
+Moreover, the asterisk denoting matrix-vector products is optional. For instance,
+both `x' = Ax` and `x' = A*x` are parsed as the linear continuous system whose
+state matrix is `A` (matrix that is supposed to be defined at the call site).
+
+**Default values.** When the dynamic equation is parsed, the variable on the
+left-hand side is interpreted as the state variable. The input variable is by
+default `u` and the noise variable is by default `w`. If we want to change the
+default name of the input variable, this can be done by adding the term
+`input: var` (or equivalently, `input=var)` where `var` corresponds to the new
+name of the input variable, eg. `@system(x' = A*x + B*v, input:v)`.
+Similarly, a noise variable is specified with `noise: var` or `noise=var`.
+
+**Exceptions.** The following exceptions and particular cases apply:
+
+- If the right-hand side has the form `A*x + B*foo` or `A*x + B*foo + c`, the
+  equation is parsed as a controlled linear (affine) system with input `foo`.
+  Note that in this case, `input` variable does not correspond to the default
+  value of `u`, but `foo` is parsed as being the input.
+
+- If the left-hand side contains a multiplicative term in the form `E*x⁺` or `E*x'`,
+  the equation is parsed as an algebraic system. In this case, the asterisk
+  `*` operator is mandatory.
+
+- Systems of the form `x' = α*x` where `α` is a scalar are parsed as linear
+  systems. The default dimension is `1`; if the system is higher-dimensional use
+  `dim`, as in `x' = 2x, dim=3`.
 
 ### Examples
 
 Let us first create a continuous linear system using this macro:
 
-```jldoctest
+```jldoctest system_macro
 julia> A = [1. 0; 0 1.];
 
 julia> @system(x' = A*x)
 LinearContinuousSystem{Float64,Array{Float64,2}}([1.0 0.0; 0.0 1.0])
 ```
-A discrete system can be defined by using `⁺`:
 
-```jldoctest
-julia> A = [1. 0; 0 1.];
+A discrete system can be defined by using `⁺`:
```

Comment:
```suggestion
A discrete system is defined by using `⁺`:
```
"can be" suggests that there are alternatives.